core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4447,6 +4450,9 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   def succeedBlocking[A](a: => A)(implicit trace: Trace): UIO[A] =
     blocking(ZIO.succeed(a))
 
+  private[zio] def succeedUnsafe[A](a: Unsafe[Any] => A)(implicit trace: Trace): UIO[A] =
```

Comment:
We could mix-in `ZIO.succeed` and `ZIO.attempt` in a Scala version-specific way, allowing Scala 3 syntax to be very clean.

Reply:
Of course, we still need the, e.g. `ZIO.succeedUnsafe` (etc) version to exist in both Scala 2 and Scala 3, for both Scala 2 code bases (which could not get away with a single unified `succeed` / `attempt`, and for our internal use.